### PR TITLE
[JENKINS-72509] show error message in progressive logs on 4xx status codes

### PR DESCRIPTION
**Pull request by mawinter69**
when accessing the log of a running build and the job gets deleted or someone removes read permissions (without discover) the log just appends the output of the call to logText/progressiveHtml which is the 404 message from Jenkins.
The same happens when one accesses the log of an agent and the agent is deleted.
This is due to only checking for 403 status and ignoring other status codes from the 4xx range.
With this change all 4xx http status code will lead to a reload and then showing a proper 404 page in mentioned cases

See [JENKINS-72509](https://issues.jenkins.io/browse/JENKINS-72509).

### Testing done
Agents:
- create an agent and connect it
- visit the log page of the agent
- in another window delete the agent
-> the page shows an error message instead of adding the 404 to the logoutput

Jobs:
- Create a job, that sleeps for a few minutes
- start the job and look at the console output (dito for consolFull)
- In a separate window delete the job
-> the page shows an error message instead of adding the 404 to the logoutput

- Create a job, that sleeps for a few minutes
- start the job
- As a different user that has job read permissions open the job log (use another browser)
- as admin user remove read permissions for that other
-> the page shows an error message instead of adding the 404 to the logoutput

### Proposed changelog entries

- JENKINS-72509, show error message in progressive logs on 4xx status codes

### Proposed upgrade guidelines

N/A

```[tasklist]
### Submitter checklist
- [x] The Jira issue, if it exists, is well-described.
- [x] The changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developers, depending on the change) and are in the imperative mood (see [examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)). Fill in the **Proposed upgrade guidelines** section only if there are breaking changes or changes that may require extra steps from users during upgrade.
- [x] There is automated testing or an explanation as to why this change has no tests.
- [ ] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadocs, as appropriate.
- [ ] New deprecations are annotated with `@Deprecated(since = "TODO")` or `@Deprecated(forRemoval = true, since = "TODO")`, if applicable.
- [ ] New or substantially changed JavaScript is not defined inline and does not call `eval` to ease future introduction of Content Security Policy (CSP) directives (see [documentation](https://www.jenkins.io/doc/developer/security/csp/)).
- [ ] For dependency updates, there are links to external changelogs and, if possible, full differentials.
- [ ] For new APIs and extension points, there is a link to at least one consumer.
```

### Desired reviewers

@mention

Before the changes are marked as `ready-for-merge`:

```[tasklist]
### Maintainer checklist
- [ ] There are at least two (2) approvals for the pull request and no outstanding requests for change.
- [ ] Conversations in the pull request are over, or it is explicit that a reviewer is not blocking the change.
- [ ] Changelog entries in the pull request title and/or **Proposed changelog entries** are accurate, human-readable, and in the imperative mood.
- [ ] Proper changelog labels are set so that the changelog can be generated automatically.
- [ ] If the change needs additional upgrade steps from users, the `upgrade-guide-needed` label is set and there is a **Proposed upgrade guidelines** section in the pull request title (see [example](https://github.com/jenkinsci/jenkins/pull/4387)).
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).
```

### CODE REVIEW

**Review by NotMyFault — APPROVED** [review]
Thanks!

File: core/src/main/resources/lib/hudson/progressive-text.js
Comment (timja) [inline core/src/main/resources/lib/hudson/progressive-text.js:41]: should this reload on server errors? it might get in a reload loop?

**Review by timja — APPROVED** [review]

**Comment by daniel-beck** [comment]
Is the diagnosis of short-lived logs a relevant use case we should continue to support, e.g., cloud agent connection logs? If so, a forced reload when it goes away makes for a worse experience. Not saying no to this, but this seems to have notable downsides compared to e.g. showing a message summarizing the error (rather than inlining the entire page, see e.g. how form validation currently does it) and offering to reload.

**Comment by mawinter69** [comment]
> Is the diagnosis of short-lived logs a relevant use case we should continue to support, e.g., cloud agent connection logs? If so, a forced reload when it goes away makes for a worse experience. Not saying no to this, but this seems to have notable downsides compared to e.g. showing a message summarizing the error (rather than inlining the entire page, see e.g. how form validation currently does it) and offering to reload.

Valid point. Changed it to print an error instead of reloading for return codes other than 403
image

**Review by NotMyFault — APPROVED** [review]
Thanks!

**Review by timja — APPROVED** [review]

**Comment by timja** [comment]
/label ready-for-merge

----

This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Pull request merged**
